File: backend/entity/post.go
package entity

import "time"

type PostStatus string

const (
	PostStatusDraft    PostStatus = "draft"
	PostStatusPublish  PostStatus = "published"
	PostStatusArchived PostStatus = "archived"
)

func (PostStatus) Values() []string {
	return []string{
		string(PostStatusDraft),
		string(PostStatusPublish),
		string(PostStatusArchived),
	}
}

type Post struct {
	ID      uint
	Title   string
	Summary *string
	Cover   *string

	User   string
	UserID uint

	Content         string
	ReadTimeMinutes uint
	ViewCount       uint

	Status PostStatus

	PublishedAt *time.Time
	CreatedAt   time.Time
	UpdatedAt   time.Time

	Tags       []PostTag
	Categories []PostCategory
}

File: frontend/app/types/post.ts
export interface PostMeta {
  id: number;
  cover: string;
  title: string;
  summary: string;
  publishedAt: string;
  updatedAt: string;
  readTimeMinutes: number;
  viewCount: number;
  tags: [];
}

export interface Post extends PostMeta {
  content: string;
}

export interface PostInput {
  cover: string;
  summary: string;
  title: string;
  content: string;
  tagIds: number[];
  status: "draft" | "published";
}
